drivers/pinctrl/pinctrl_npcm.c

```diff
@@ -0,0 +1,383 @@
+	NPCM_PIN_PERIPH_PUPD_GROUP_RDATA(NPCM_PINMUX_ID(node_id, id),                              \
+					 NPCM_PIN_PERIPH_PUPD_SUP(node_id, pupd_sup),              \
+					 NPCM_PIN_PERIPH_PUPD_REG0(node_id, pupd_reg),             \
+					 NPCM_PIN_PERIPH_PUPD_TYPE0(node_id, pupd_type),           \
+					 NPCM_PIN_PERIPH_PUPD_REG1(node_id, pupd_reg),             \
+					 NPCM_PIN_PERIPH_PUPD_TYPE1(node_id, pupd_type))
+
+#define NPCM_PIN_PERIPH_PUPD_INIT(node_id, prop)                                                   \
+	COND_CODE_1( \
+		DT_PROP(node_id, pinmux_group_def), \
+		(COND_CODE_1( \
+			DT_NODE_HAS_PROP(node_id, pupd_reg), \
+			(COND_CODE_1( \
+				DT_PROP(node_id, pupd_config_extend), \
+				(NPCM_PIN_PERIPH_PUPD_GROUP_EXT_INIT(node_id)), \
+				(NPCM_PIN_PERIPH_PUPD_GROUP_INIT(node_id)) \
+				) \
+			), \
+			()) \
+		), \
+		())
+
+static const struct npcm_periph_pupd_cfg npcm_periph_pupd_table[] = {
+	DT_FOREACH_CHILD_VARGS(DT_DRV_INST(0), NPCM_PIN_PERIPH_PUPD_INIT)};
+static const int npcm_periph_pupd_table_size = ARRAY_SIZE(npcm_periph_pupd_table);
+
+/*
+ * Pin voltage level definitions.
+ *
+ * The table is used to check the current pin voltage level. Not all alt functions in a pin have
+ * low-voltage support. The recorded status is only for supported alt functions.
+ */
+struct npcm_pin_lv_cfg {
+	uint16_t id: 7;      /* Pin number */
+	uint16_t lv_reg: 11; /* Low-voltage config reg field */
+	uint16_t lv_sup: 6;  /* The bit N value means group N supports low-voltage level. */
+	uint16_t level: 1;   /* 0 = normal 3.3V, 1 = low-voltage 1.8V */
+} __packed;
+
+#define NPCM_PIN_LV_GROUP_RDATA(pin, reg, sup, lv)                                                 \
+	{                                                                                          \
+		.id = pin,                                                                         \
+		.lv_reg = reg,                                                                     \
+		.lv_sup = sup,                                                                     \
+		.level = lv,                                                                       \
+	},
+#define NPCM_PIN_LV_REG(node_id, prop)       DT_PROP_BY_IDX(node_id, prop, 0)
+#define NPCM_PIN_LV_SUP(node_id, prop)       DT_PROP_BY_IDX(node_id, prop, 0)
+#define NPCM_PIN_LV_LOW_LEVEL(node_id, prop) DT_PROP_BY_IDX(node_id, prop, 0)
+
+#define NPCM_PIN_LV_INIT(node_id, prop)                                                            \
+	COND_CODE_1( \
+		DT_PROP(node_id, pinmux_group_def), \
+		(COND_CODE_1( \
+			DT_NODE_HAS_PROP(node_id, lv_reg), \
+			(NPCM_PIN_LV_GROUP_RDATA( \
+				NPCM_PINMUX_ID(node_id, id), \
+				NPCM_PIN_LV_REG(node_id, lv_reg), \
+				NPCM_PIN_LV_SUP(node_id, lv_sup), \
+				NPCM_PIN_LV_LOW_LEVEL(node_id, lv_def) \
+				) \
+			), \
+			()) \
+		), \
+		())
+
+static struct npcm_pin_lv_cfg npcm_pin_lv_table[] = {
+	DT_FOREACH_CHILD_VARGS(DT_DRV_INST(0), NPCM_PIN_LV_INIT)};
```

Comment:
use `DT_INST_*` APIs, DT_DRV_INST() should be avoided.

Reply:
Fixed.